### DISCUSSION TITLE
Can't log-in because can't get email on DigitalOcean

### DISCUSSION THREAD

**Discussion by sfmoa** [mozilla/hubs#3134] [Hubs Cloud Questions] [answered]
Have Hubs installed in DO, but no email is sending so can't log-in. I'm trying Sendgrid and also Zohomail which is the personal email.

Following guides, I telnet into the smtp servers in the DO droplet, but they hang with no successful connection. Can someone try these commands on DO and share what happens?

telnet smtp.zoho.com 2525
Trying 136.143.186.56...

telnet smtp.sendgrid.net 2525
Trying 167.89.118.51...

 telnet smtp.sendgrid.net 587
Trying 169.45.89.186...

telnet smtp.zoho.com 587
Trying 136.143.186.56...

 telnet smtp.zoho.com 465
Trying 136.143.186.56...
Connected to smtp.zoho.com.
Escape character is '^]'.

telnet smtp.sendgrid.net 465
Trying 169.45.89.186...
Connected to smtp.sendgrid.net.
Escape character is '^]'.

Even though for the last two case, it says 'Connected', this is not a success neither because after typing in further log-in commands, I get disconnected.

I'm aware DO won't unblock until two months later. But since Zohomail and Sendgrid are not part of DO, they shouldn't be blocked right?

**Comment by robinkwilson** [comment, marked as answer]
I've never used telnet to configure the SMTP settings. 

For DO, you'll need to:
1. ssh into your box
2. Run `/opt/polycosm/setup.sh`
3. Setup your SMTP settings via the prompts
4. Try to login to your hub / see if you receive an email

My DO SMTP settings were (from memory):
smtp.sendgrid.net
587
{{main domain}}.com - NOT the default mail.{{main domain}}.com

Let me know if this works for you!

**Reply by sfmoa** [reply]
thanks. One thing I'm confused about is, for Sendgrid, it wants you to make an api key. Then, 'apikey'  becomes the username field's value, and the generated long api key string becomes the password field's value. Then, which email address is actually used to send? Usually name@myDomain.com goes into the username field, and myPassWord string goes into the password field

**Reply by robinkwilson** [reply]
The one you specify, for example I use `{{main domain}}.com` as the field, `noreply@{{main domain}}.com` will be the email.

**Reply by shapesim** [reply]
Thanks

[main domain.com] is hosted with DO droplet
but first.last@[main domain.com] is hosted by Zohomail, not some mail server in DO droplet. Is this the problem?

So I don't know how Sendgrid creates a noreply@[main domain.com] because it doesn't control Zohomail's servers.


My settings are like this while setting up:

SMTP server: smtp.sendgrid.net
? SMTP port: 587
? SMTP User: apikey
? SMTP Password: [hidden]
? From domain for emails: main domain.com

And I use a gmail address as the admin email address

Administrator email address: first.last@gmail.com
? Your hub's domain name (eg myhub.com or hub.mycompany.com): main domain.com
? Your hub's shortlinks domain (eg myhub.link): main domain.net

**Reply by robinkwilson** [reply]
Did this work for you?

Zoho may be a problem, does your main domain have an MX record pointing to Zoho? -- I'm not sure sendgrid needs to configure that on the main domain.

**Reply by sfmoa** [reply]
edit:
It's working now.
For some reason port 587 seems unblocked now whereas it was blocked yesterday. Hopefully it stays open

Thanks so much

Ignore rest below




Was able to send email from the main domain successfully in Sendgrid.com's by making a marketing campaign. So I think that means the domain is verified properly. 

I think based on the telnet commands, it's DO that's blocking 587 and 2525 and thus causing email not be sent? This is a new DO account so can't yet request DO to unblock any ports. But, I think you and other users have had success even as new accounts in DO?

Is there a log in Hubs Cloud that can be checked?

Thanks

**Reply by robinkwilson** [reply]
Wonderful! I'm glad it's working now! :D

**Reply by jjanisheck** [reply]
I was running into the exact same problems.  I was able to resolve with adding a few additional subdomain CNAME pointers back to SendGrid.  That was the piece I was missing.

https://app.sendgrid.com/settings/sender_auth
